### PR TITLE
chore: set size_bytes to source volume size and add different allocation sizes

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

**Review by hrudaya21 — APPROVED** [review]

File: control-plane/csi-driver/src/bin/controller/controller.rs [hunk 1]
```diff
@@ -818,7 +818,7 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
 
 fn snapshot_to_csi(snapshot: models::VolumeSnapshot) -> Snapshot {
     Snapshot {
-        size_bytes: snapshot.state.size as i64,
+        size_bytes: 0,
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/controller/controller.rs]: Please add a comment stating we should come back to this once we know what this size should be exactly.

**Review by tiagolobocastro — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
@tiagolobocastro @hrudaya21 Will update this PR with the changes merged in the release/2.3 once @mtzaurus's changes are merged in develop.

**Comment by Abhinandan-Purkait** [comment]
@hrudaya21 @mtzaurus Can you please cherry-pick this to develop https://github.com/openebs/mayastor/pull/1448, it would be good to have these rpc changes in before any more rpc changes are done.

**Comment by hrudaya21** [comment]
> @hrudaya21 @mtzaurus Can you please cherry-pick this to develop [openebs/mayastor#1448](https://github.com/openebs/mayastor/pull/1448), it would be good to have these rpc changes in before any more rpc changes are done.

https://github.com/openebs/mayastor/pull/1451
https://github.com/openebs/mayastor-api/pull/74

Above PR is raised in develop.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**